### PR TITLE
Rework of psq tables

### PR DESCRIPTION
**Pull request by mcostalba**
Original patch from Lucas, here a bit reworked and split in 2 patches because each one addresses a  different target.

See single commit messages for details.

### CODE REVIEW

**Comment by lucasart** [comment]
Looks good. In the end, the only difference between your version and mine is that I used a 3 dimensional array `PSQ[pieceType][rank][edgeDistance]` where you used a 2 dimensional array `PSQ[pieceType][4*rank+edgeDistance]`.

**Comment by zamar** [comment]
approved...

File: src/psqt.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  {
+      PieceValue[MG][make_piece(BLACK, pt)] = PieceValue[MG][pt];
+      PieceValue[EG][make_piece(BLACK, pt)] = PieceValue[EG][pt];
+
+      Score v = make_score(PieceValue[MG][pt], PieceValue[EG][pt]);
+
+      for (Square s = SQ_A1; s <= SQ_H8; ++s)
+      {
+          // Map SQ_A5 -> SQ_A4, SQ_F8 -> SQ_F1
+          int ss = s / 8 * 4 + (s % 8 > 3 ? 3 - s % 4 : s % 4);
```
Comment (ceebo) [inline src/psqt.cpp]: Here is an alternative formula and comment:

```
          // Flip to the left half of the board and subtract 4 for each rank
          int ss = (file_of(s) < FILE_E ? s : s ^ 7) - 4 * rank_of(s);
```

Comment (mcostalba) [inline src/psqt.cpp]: Thanks, it is a more readable formula: applied.

**Comment by glinscott** [comment]
Looks good!

**Pull request merged**